### ISSUE TITLE
[REVIEW]: The DynaGUI package

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1942]
**Submitting author:** @benjaminbolling (<a href="http://orcid.org/0000-0002-6650-5365">Benjamin Edward Bolling</a>)
**Repository:** <a href="https://github.com/benjaminbolling/DynaGUI" target ="_blank">https://github.com/benjaminbolling/DynaGUI</a>
**Version:** v1.0.1
**Editor:** @majensen
**Reviewer:** @Chilipp, @pythonpanda2
**Archive:** <a href="https://doi.org/10.5281/zenodo.4396227" target="_blank">10.5281/zenodo.4396227</a>

## Status

[![status](https://joss.theoj.org/papers/eede632738a83ce2b21bd85fe039ed45/status.svg)](https://joss.theoj.org/papers/eede632738a83ce2b21bd85fe039ed45)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/eede632738a83ce2b21bd85fe039ed45"><img src="https://joss.theoj.org/papers/eede632738a83ce2b21bd85fe039ed45/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/eede632738a83ce2b21bd85fe039ed45/status.svg)](https://joss.theoj.org/papers/eede632738a83ce2b21bd85fe039ed45)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/benjaminbolling/DynaGUI" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@Chilipp & @pythonpanda2, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @majensen know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/benjaminbolling/DynaGUI">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@benjaminbolling) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/benjaminbolling/DynaGUI">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@benjaminbolling) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @Chilipp, @pythonpanda2 it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1107/S1600577514011503 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01942/joss.01942/10.21105.joss.01942.pdf)

**Comment by pythonpanda2** [comment]
https://github.com/benjaminbolling/DynaGUI/issues/2#issue-538141258

**Comment by pythonpanda2** [comment]
https://github.com/benjaminbolling/DynaGUI/issues/3

**Comment by pythonpanda2** [comment]
https://github.com/benjaminbolling/DynaGUI/issues/4

**Comment by majensen** [comment]
Hi @Chilipp - just wanted to ping you and ask if you are still willing to review this submission around 20 Jan. Thanks!

**Comment by Chilipp** [comment]
Yes @majensen, I will do so.

**Comment by majensen** [comment]
Hi @pythonpanda2 and @Chilipp  -- looking forward to your reviews. Thank you very much for your time on this!

**Comment by pythonpanda2** [comment]
@majensen 

Looking forward to the response to the  queries.

**Comment by Chilipp** [comment]
Dear @majensen and @benjaminbolling. My apologies for the delay and thank you very much for your patience! I am done with my review (so far) and note a high number of issues to work on concerning the DynaGUI package. As these are so many, I did not want to create an issue for each individual point, but rather want to give @benjaminbolling the chance to comment on each of these.

My general feeling is, that this repository is a collection of scripts that have a massive lack in documentation. There are no installation instructions, there is no quick start guide (not even a README in the repository), and there are only explanations about the GUI widgets, but not even about how to start the GUI. Please let me know @benjaminbolling and @majensen if you want me to open issues for the individual points listed below.

### General checks


> **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

I recommend to rename GNU_Licence.txt to LICENSE, as this is common for software packages. But this is not a blocking issue, of course.

> **Contribution and authorship:** Has the submitting author (@benjaminbolling) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

I cannot judge this @majensen as the commit history is not present in the repository. Don't know how to handle this, but from the paper itself it seems like some others have been involved from [Tavares et. al](https://doi.org/10.1107/S1600577514011503)

### Functionality

> **Installation:** Does installation proceed as outlined in the documentation?

I could not find any installation instructions. Please create a README.md and provide some guidance for the user.

> **Functionality:** Have the functional claims of the software been confirmed?

I couldn't because I did not install it.


### Documentation

There exists a PDF-Document in the repository ([DynaGUI_documentation.pdf](https://github.com/benjaminbolling/DynaGUI/blob/master/DynaGUI_documentation.pdf)) which serves, I guess as the documentation. So I will judge based on this document.

> **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.

There is a System Requirements section, but no installation instructions.

> **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).

No, there is no example usage. There are some screenshots of the different widgets, but I could not find any executable code in the documentation to verify the functionality of the package.

> **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?

As there are only explanations of the various images, no, it is not explained well enough. There should be a piece of code somewhere stating how to generate and open the different widgets. How to establish a connection to the server, etc. The code also contains almost no documentation/comments.

> **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?

I could not find any tests associated with the repository.

> **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

No, there are no community guidelines.

### Software paper

> **State of the field:** Do the authors describe how this software compares to other commonly-used packages?

I could not find any comment on that in the paper.

> **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

The package also depends on matplotlib and numpy, which need to be cited.

**Comment by benjaminbolling** [comment]
Dear Philipp,

Many thanks for your review. I will rewrite the code for the package shortly so that it does not consist solely “of a bunch of scripts” but instead of a whole package and I will include an installation method with instructions, and instructions / a README file with clear instructions. 

I will be more clear in the list of contributors, which have been included in the acknowledgments section. I will cite MatPlotLib and NumPy, and compare with other similar software.

Thank you for your time and patience.

Kind regards 
Benjamin

**Comment by Chilipp** [comment]
Great to hear @benjaminbolling, I am looking forward to it. Just tell me, when you are ready for another round of review or if there is anything else.

Thanks for your work on this!

**Comment by majensen** [comment]
@benjaminbolling please remember also to review @pythonpanda2 's questions that were created as issues on the DynaGUI repo.  thanks!

**Comment by majensen** [comment]
Hi @benjaminbolling - how are things proceeding?

**Comment by benjaminbolling** [comment]
Hi,

I am rewriting some code and adding in-code explanations which can assist you during the reviewal. I am sorry for taking so much time, but I will let you know when it is ready together with a clear manual and installation routine such that the reviewal will go smoothly. I have also been busy with a lot of work, but I am working on this as much as I can.

Thanks for the patience!

**Comment by majensen** [comment]
Thanks for the update @benjaminbolling -- no worries

**Comment by labarba** [comment]
👋 @benjaminbolling — since it's been 3 weeks since your last comment here, would you like to tell us an estimated time for your work-in-progress? We can set a reminder and/or add the `paused` label.

**Comment by majensen** [comment]
@benjaminbolling can you give us an idea of your timeline, per @labarba 's comment? I will see if I can reach you via an alternative channel as well.

**Comment by majensen** [comment]
Thanks @benjaminbolling for touching base with me on the alternative channel. We'll add the "paused" label to this submission for now. Feel free to pick it back up with a post here.

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by majensen** [comment]
@whedon remind @majensen in 2 weeks

**Comment by whedon** [comment]
Reminder set for @majensen in 2 weeks

**Comment by arfon** [comment]
@majensen - looks like Whedon forgot to remind you :-)

**Comment by majensen** [comment]
Well that's a data point @arfon !

**Comment by majensen** [comment]
@benjaminbolling just want to check in here with you. Let me know how it's going.

**Comment by benjaminbolling** [comment]
Det all,

Thank you for being so patient! I am ensuring there are no bugs in the code now, finishing some additional features of it and expecting it to be ready with instructions and a short article before the end of this month.

Again, thank you for the patience!

Kind regards
Benjamin

**Comment by majensen** [comment]
Sounds good @benjaminbolling  - I will unpause us.

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #1942 with the following error: 

 Error reading bibliography ./paper.bib (line 31, column 1):
unexpected "@"
expecting space, ",", white space or "}"
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #1942 with the following error: 

 Error reading bibliography ./paper.bib (line 32, column 14):
unexpected "a"
expecting white space, "#", "," or "}"
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01942/joss.01942/10.21105.joss.01942.pdf)

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01942/joss.01942/10.21105.joss.01942.pdf)

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01942/joss.01942/10.21105.joss.01942.pdf)

**Comment by benjaminbolling** [comment]
Dear all,
@majensen @arfon @Chilipp 
thanks for the enormous patience. I have now updated the package, attached a README file with installation instructions and how to launch it, a documentation on how to use the applications in the package, and also updated the article itself.

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01942/joss.01942/10.21105.joss.01942.pdf)

**Comment by majensen** [comment]
@Chilipp @pythonpanda2 - Hope to see your comments within the next week or so. Thank @benjaminbolling for all this work.

**Comment by Chilipp** [comment]
Dear @majensen and @benjaminbolling, thank you for the update! I am looking forward to have another look at it, but I am on vacation until June 12th. I expect to finish the review by June 21st.

**Comment by pythonpanda2** [comment]
@majensen @benjaminbolling 

I am starting a new job this week. Hopefully i would try to get back to this sometime next week.

**Comment by majensen** [comment]
Hi @pythonpanda2 - I hope the new job is going well. Do you have a sense of when you might have the opportunity to look at @benjaminbolling 's work? Thanks very much  -- @majensen

**Comment by pythonpanda2** [comment]
@majensen Apologies. I will be try to complete the review over the weekend.

**Comment by pythonpanda2** [comment]
https://github.com/benjaminbolling/DynaGUI/issues/3#issuecomment-647886871

**Comment by Chilipp** [comment]
>  I expect to finish the review by June 21st.

Dear @majensen and @benjaminbolling. I have to apologize but something urgent forced me to change my plans and I won't have the possibility to review this within this week. I have to postpone it until next week and hope, this does not cause problems.

**Comment by majensen** [comment]
@Chilipp - no problem; thanks for the heads up. Will touch base next week-

**Comment by majensen** [comment]
@whedon remind @Chilipp in 7 days

**Comment by whedon** [comment]
Reminder set for @Chilipp in 7 days

**Comment by pythonpanda2** [comment]
There are already some very well rounded points raised by @Chilipp in the last round  . I would wait to see if some of those points have been addresses to their satisfaction  (To avoid repetitions) and   i will stick to the one issue I have raised. 

In addition, the funding sources should be duly acknowledged.

**Comment by whedon** [comment]
:wave: @Chilipp, please update us on how your review is going.

**Comment by Chilipp** [comment]
Hi @benjaminbolling,

my apologies for the delay and thank you very much @benjaminbolling for your ongoing work on this!

I have the feeling that most of my points [from my comment above](https://github.com/openjournals/joss-reviews/issues/1942#issuecomment-577582216) have not been addressed (or at least I could not find them). As it takes me quite a while to go through everything, please address the individual issues in the future that I reference down below, or cite my comments and answer within this thread. 
I apologize if I just did not find the information on some of the issues mentioned below. If so, please point me individually to the correct place where this is answered from your perspective or ask if there is need for clarification.


### General checks

> **Contribution and authorship:**

- [ ] @majensen, as stated above, I cannot judge this as I don't see the commit history. So this is something you have to decide

### Functionality

> **Installation:** Does installation proceed as outlined in the documentation?

- [ ] As already pointed out by @pythonpanda2 in benjaminbolling/DynaGUI#3, you should to add the requirements to the installation instruction, etc., see also benjaminbolling/DynaGUI#6
- [ ] You should rewrite the README.md file to make it properly markdown-formatted benjaminbolling/DynaGUI#5

> **Functionality:** Have the functional claims of the software been confirmed?

- [ ] I was able to start it using the environment file from benjaminbolling/DynaGUI#6, but as there is no example usage, I could not really test it (see benjaminbolling/DynaGUI#7)

### Documentation

> **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.

- [ ] I did add an environments.yml file in benjaminbolling/DynaGUI#6, but we would still need to add proper installation instructions to the README file (or somewhere else)

> **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).

- [ ] Could you point me to the place where this has been added or where I can find something about it? see benjaminbolling/DynaGUI#7

> **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?

- [ ] There is no API documentation see benjaminbolling/DynaGUI#8

> **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?

- [ ] I could not find any tests associated with the repository.

> **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

- [ ] No, there are still no community guidelines, see benjaminbolling/DynaGUI#9

### Software paper

> **State of the field:** Do the authors describe how this software compares to other commonly-used packages?

- [ ] @benjaminbolling, could you please point me to the place where this is shown as I could not find it

> **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

- [ ] you are still not citing matplotlib and numpy in your paper. see https://www.scipy.org/citing.html#numpy and https://matplotlib.org/3.1.1/citing.html

**Comment by majensen** [comment]
@Chilipp thanks for this summary - @benjaminbolling can you address these issues? Just let me know an approximate timeline here. Thanks!

**Comment by benjaminbolling** [comment]
Thank you @Chilipp, I will be addressing these issues in the coming two weeks in the evenings. @majensen Thus, we can set a deadline for next Friday, July 17th.

**Comment by majensen** [comment]
@whedon remind @majensen 17 July 2020

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by majensen** [comment]
@whedon remind @majensen in 10 days

**Comment by whedon** [comment]
Reminder set for @majensen in 10 days

**Comment by benjaminbolling** [comment]
@Chilipp : I have updated the code a bit to address issue#4: To decrease the dependency of PyTango. All codes now enables user to select between the packages "Tango" (using PyTango), "EPICS" (using PyEPICS), "Finance" (using Pandas Datareader) and "Randomizer" (for testing and demonstration purposes using random values). Other packages used for obtaining values from sources can be added in later stages with ease. I will address each issue in their own reference.

As for "Contribution and authorship", I did all of the code except for the original code for the "2D Spectrogram" part which I referenced in both the paper.md and the documentation file under Acknowledgements. If this is incorrect or if you need me to change something there, let me know.

Again, thanks for the patience!

**Comment by benjaminbolling** [comment]
I have addressed issues 3, 4, 5,7, 8, 9. 

Installation instructions: Added to the README file. 

State of the field: I am unaware of currently existing Python-based package for constructing dynamic GUI:s (without any programming), and I have added this also to the README file. 

Left to do: 
- Create some automated tests/steps described so that the functionality of the software can be verified
- Add API documentation

The last part might sadly take a little bit longer time, but I am working on it.

**Comment by benjaminbolling** [comment]
@Chilipp @pythonpanda2 @majensen : I have written a test procedure, let me know if you find it sufficient or not. The API documentation will, as stated before, take a little while (ca. a week). You can start testing it anyways if you'd want to, as it seems to me as if all other issues and comments have now been addressed. Thank you all for the patience and the comments, and have a nice weekend!

**Comment by majensen** [comment]
@benjaminbolling thanks for this work. A tip, if you rename your test.txt to test.md, then GitHub will render it in a nice form.

@pythonpanda2 - based on the work done, can you check off the items in your review that are satisfied?

thanks everyone

**Comment by benjaminbolling** [comment]
thanks @majensen, I have renamed test.txt to test.md as you suggested.

**Comment by majensen** [comment]
@pythonpanda2 please let us know here if you are still able to review this work. Thanks --@majensen

**Comment by pythonpanda2** [comment]
Sorry  for the delay. In between the peak review season now. I will have a looking this week.

**Comment by Chilipp** [comment]
thanks @benjaminbolling! I'll have a look at it within this week

**Comment by pythonpanda2** [comment]
@majensen  @benjaminbolling  I have closed this issue . https://github.com/benjaminbolling/DynaGUI/issues/2

**Comment by pythonpanda2** [comment]
@majensen  @benjaminbolling  It would be good to see if we can resolve this : https://github.com/benjaminbolling/DynaGUI/issues/3

**Comment by majensen** [comment]
@benjaminbolling Have you been able to take care of benjaminbolling/DynaGUI#3 per @pythonpanda2 ?

**Comment by majensen** [comment]
@Chilipp @pythonpanda2 @benjaminbolling I would like to take stock of where we are in this review. Could reviewers please make sure that the items they are satisfied with are checked off in their reviewer forms? What are the next steps @benjaminbolling for finishing with reviewer suggestions? Thank you all-

**Comment by Chilipp** [comment]
> Could reviewers please make sure that the items they are satisfied with are checked off in their reviewer forms?

Everything's up-to-date from my side @majensen

**Comment by kthyng** [comment]
@Chilipp and @pythonpanda2 Hi! I am the rotating associate editor in chief here this week trying to help ping stale issues. @Chilipp I see you said just above that you are satisfied with the submission, but you have a lot of unchecked boxes above. Could you either check the boxes or state what are the next issues?
@pythonpanda2 How is your review coming along?

Thanks all!

**Comment by Chilipp** [comment]
Hello @kthyng! There seems to be a misunderstanding. No, the items are not yet resolved. My last comment just states that the open points I mentioned are still open. @benjaminbolling just made some changes yesterday and I'll review them as soon as possible

**Comment by kthyng** [comment]
@Chilipp Ok excellent! Thanks for your response and glad things are working out ok.

**Comment by majensen** [comment]
@kthyng thanks for checking in. Things have started to proceed over at https://github.com/benjaminbolling/DynaGUI/issues/3#issuecomment-704386598

**Comment by majensen** [comment]
@benjaminbolling any progress to report? thanks

**Comment by benjaminbolling** [comment]
@majensen Sorry for the delay in my reply, I have a lot of work nowadays during the pandemic. 

I have replied to [Issue #3](https://github.com/benjaminbolling/DynaGUI/issues/3).

I will deal with [Issue #7](https://github.com/benjaminbolling/DynaGUI/issues/7) this evening.

I am close to finished with the API documentation for [Issue #8](https://github.com/benjaminbolling/DynaGUI/issues/8), and will post here when it is finished.

Again, thanks for all patience and all your efforts put into this!

**Comment by benjaminbolling** [comment]
I have now dealt with [Issue #7](https://github.com/benjaminbolling/DynaGUI/issues/7), let's see if this will do. 🙂 

Continuing with [Issue #8](https://github.com/benjaminbolling/DynaGUI/issues/8) now.

**Comment by majensen** [comment]
@benjaminbolling @Chilipp @pythonpanda2  - as we're coming up on the one year anniversary of the submission, I thought I would see if we're still on track. I think @benjaminbolling  is making some progress. @Chilipp I think you are still with us. @pythonpanda2 can you ping back on this thread as to whether you are still following this review and able to pick it up again?
thanks all

**Comment by benjaminbolling** [comment]
Good morning all, and thanks again. Can you confirm what is left to be done? I think I am ready to push the changes in terms of the last part of the API documentation, [Issue #8](https://github.com/benjaminbolling/DynaGUI/issues/8).

Let me know what else is left to do. Thanks again all, and sorry for taking so much time.

**Comment by Chilipp** [comment]
Thanks for your work on this @benjaminbolling! @majensen I will go through it again by the end of this week. My apologies for the delay as my workload is pretty high at the moment and it takes some time for me to get into this again.

**Comment by pythonpanda2** [comment]
> @benjaminbolling @Chilipp @pythonpanda2 - as we're coming up on the one year anniversary of the submission, I thought I would see if we're still on track. I think @benjaminbolling is making some progress. @Chilipp I think you are still with us. @pythonpanda2 can you ping back on this thread as to whether you are still following this review and able to pick it up again?
> thanks all

I think closed the last issued i had opened. But i will go through the checklist once more by early next week.

**Comment by majensen** [comment]
Thanks a lot everybody- we are closing in!

**Comment by Chilipp** [comment]
Thanks for your patience and apologies again for the delay @majensen and @benjaminbolling! 

I went through all open points from the checklist for JOSS and understand them as fulfilled. So I recommend this paper to be accepted.

**Comment by pythonpanda2** [comment]
I have reviewed the checklist. I would  recommend the DynaGUI research software article  for publication in JOSS.

**Comment by majensen** [comment]
Thanks everyone. I surely appreciate all the work and tenacity on this review. I will do some editorial checks on the paper, which may result in minor pull request. Once that is complete, @benjaminbolling, I'll ask you to create a public archive of the repo using Zenodo or similar. After that we can summon the associate editors-in-chief by making our recommendation.

**Comment by benjaminbolling** [comment]
Thank you @majensen!

**Comment by majensen** [comment]
Holiday has delayed me - hope to get to this this week. Thanks all

**Comment by majensen** [comment]
@benjaminbolling I promise to get this done this weekend. Sorry about the delay!

**Comment by majensen** [comment]
@whedon check references

**Comment by majensen** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1107/S1600577514011503 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- 10.1109/mcse.2011.37 may be a valid DOI for title: The NumPy Array: A Structure for Efficient Numerical Computation

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.01942/joss.01942/10.21105.joss.01942.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.01942/joss.01942/10.21105.joss.01942.pdf) :page_facing_up: :point_left:

**Comment by benjaminbolling** [comment]
thank you @majensen!

**Comment by benjaminbolling** [comment]
I saw there was a missing DOI for the NumPy array, so I added this reference to paper.bib.

**Comment by benjaminbolling** [comment]
@whedon check references

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1107/S1600577514011503 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.01942/joss.01942/10.21105.joss.01942.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.01942/joss.01942/10.21105.joss.01942.pdf) :page_facing_up: :point_left:

**Comment by majensen** [comment]
@benjaminbolling  you'll see a PR with a few scattered edits. 
The paper is missing "Figure 1", referred to in the text - not present in the repo.
Thanks

**Comment by benjaminbolling** [comment]
@majensen Done!

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.01942/joss.01942/10.21105.joss.01942.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.01942/joss.01942/10.21105.joss.01942.pdf) :page_facing_up: :point_left:

**Comment by benjaminbolling** [comment]
Thanks a lot for the suggested edits, they have been merged.

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.01942/joss.01942/10.21105.joss.01942.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.01942/joss.01942/10.21105.joss.01942.pdf) :page_facing_up: :point_left:

**Comment by majensen** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1107/S1600577514011503 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by majensen** [comment]
Ok @benjaminbolling -- I think we are on our way to the finish. The last thing to do before I call upon Those Who Must Not Be Named (the associate editors in chief) is to create an open archive of the Github repo. Zenodo.org is a good place for this - have a look at https://github.com/openjournals/joss-reviews/issues/1839#issuecomment-560048729 for a brief tutorial.
The main detail here is to name the archive the same as the paper ("The DynaGUI Package") and for the archive to have the same author as the paper. You'll have an archive version and a DOI -- please report those back in this thread.

**Comment by benjaminbolling** [comment]
Dear @majensen, sorry for the delay. There was simply too much to do before the holidays.

Archive version: Version 1
DOI: 10.5281/zenodo.4393589

**Comment by majensen** [comment]
@whedon set archive as 10.5281/zenodo.4393589

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by majensen** [comment]
@whedon set 10.5281/zenodo.4393589 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4393589" target="_blank">10.5281/zenodo.4393589</a> is the archive.

**Comment by majensen** [comment]
@whedon set "Version 1" as version

**Comment by whedon** [comment]
OK. "Version 1" is the version.

**Comment by majensen** [comment]
@whedon recommend accept

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by majensen** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1107/S1600577514011503 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2006

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2006, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@benjaminbolling  - do you mean for the version name to be "Version 1" as opposed to "v1.0" or "1.0"?

(Also, as an FYI, the link at the end of the README to the license doesn't work, and your use of ":s" in the README is also confusing, as in "GUI:s" and "BPM:s" - if you mean these to be plurals, they should just be "GUIs" and "BPMs", I think, and if they are intended to be possessive, which I'm fairly sure you don't, "GUI's" and "BPM's" would be correct)

**Comment by danielskatz** [comment]
In the paper, the fact that the first paragraph has some text written in the first person (I created), while the rest of the paper is written in the third paragraph (The author wants to thank) is a little confusing. Maybe change the first part to third person too?

Otherwise, this looks ready to accept.

**Comment by benjaminbolling** [comment]
Dear @danielskatz, thank you for your comments. I have change the use of ":s" to just "s" (when it was used to describe objects in plural). Also, the first paragraph in the user guide is rewritten to third paragraph and the acknowledgements has been updated to sound less confusing.

In accordance with this, a new version of the files have been archived on Zenodo with a new DOI.

**Comment by benjaminbolling** [comment]
@whedon set 10.5281/zenodo.4396035 as archive

**Comment by whedon** [comment]
I'm sorry @benjaminbolling, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by benjaminbolling** [comment]
@majensen : Please update Zenodo archive to:

Archive version: v1.0
DOI: 10.5281/zenodo.4396035

Thank you!

**Comment by danielskatz** [comment]
@whedon set v1.0 as version

**Comment by whedon** [comment]
OK. v1.0 is the version.

**Comment by danielskatz** [comment]
@whedon set 10.5281/zenodo.4396035 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4396035" target="_blank">10.5281/zenodo.4396035</a> is the archive.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1107/S1600577514011503 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2007

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2007, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@benjaminbolling  - I don't see any changes in the paper - the last update seems to be 14 days ago

https://github.com/benjaminbolling/DynaGUI/blob/master/paper.md

**Comment by benjaminbolling** [comment]
I am sorry, I missed to commit it. It has been committed now.

**Comment by benjaminbolling** [comment]
Removed .git from zip file and updated paper.md in the archived file.

Please update Zenodo archive to:

Archive version: v1.0.1
DOI: 10.5281/zenodo.4396227

**Comment by danielskatz** [comment]
Thanks - there's no need to update the repository for a change in the paper, but it's also fine to

**Comment by danielskatz** [comment]
@whedon set v1.0.1 as version

**Comment by whedon** [comment]
OK. v1.0.1 is the version.

**Comment by danielskatz** [comment]
@whedon set 10.5281/zenodo.4396227 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4396227" target="_blank">10.5281/zenodo.4396227</a> is the archive.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1107/S1600577514011503 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2008

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2008, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1343671383335821326) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2009
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01942](https://doi.org/10.21105/joss.01942)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by benjaminbolling** [comment]
Amazing! A big thank you to everyone for all the time and effort you put into this!

**Comment by danielskatz** [comment]
It's not quite done yet, but close - at least, the paper doesn't appear in the link.

Once it does, I'll close this issue

**Comment by majensen** [comment]
Fantastic @benjaminbolling! Congratulations! Thanks for staying with us to the end @pythonpanda2 and @Chilipp. Really happy to see this come to fruition.

Cheers to @danielskatz, always appreciated.

**Comment by danielskatz** [comment]
Ok, the paper is now there for me.

Congratulations to @benjaminbolling (Benjamin Edward Bolling)!!

And thanks to @Chilipp and @pythonpanda2 for reviewing, and @majensen for editing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01942/status.svg)](https://doi.org/10.21105/joss.01942)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01942">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01942/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01942/status.svg
   :target: https://doi.org/10.21105/joss.01942
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01942/status.svg)](https://doi.org/10.21105/joss.01942)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
